### PR TITLE
Feature/white238/systemcmake

### PR DESCRIPTION
**Pull request by white238**
This pull request kinda went of the rails... but here's what it does:

* Update Toss3 compilers
  - clang@4.0.0 to clang@9.0.0 with an updated libstdc++ (gcc 8.1.0)
  - intel@19.0.0 to intel@19.0.4 with an updated libstdc++ (gcc 8.1.0)
* Fix our Serac, Axom, and Conduit Spack packages to honor Spack's global compiler flags (these will be pushed upstream as soon as the needed Spack bugfix branch gets merged)
* Use system CMake 3.14.5 on Toss3
* Quiet a CMake warning about HDF5

### CODE REVIEW

**Comment by white238** [comment]
Quartz host-configs coming soon hopefully

File: scripts/uberenv/spack_configs/config.yaml [hunk 1]
```diff
@@ -55,7 +55,7 @@ config:
 
   # Cache directory for miscellaneous files, like the package index.
   # This can be purged with `spack purge --misc-cache`
-  misc_cache: ~/.spack/cache
+  misc_cache: .spack/misccache
```
Comment (jamiebramwell) [inline scripts/uberenv/spack_configs/config.yaml:58]: Thank you!!

File: scripts/uberenv/packages/serac/package.py [hunk 2]
```diff
@@ -209,6 +213,26 @@ def hostconfig(self, spec, prefix, py_site_pkgs_dir=None):
         cfg.write(cmake_cache_entry("CMAKE_C_COMPILER", c_compiler))
         cfg.write(cmake_cache_entry("CMAKE_CXX_COMPILER", cpp_compiler))
 
+        # use global spack compiler flags
```
Comment (jamiebramwell) [inline scripts/uberenv/packages/serac/package.py:216]: Thank you for figuring this out!

File: scripts/uberenv/packages/serac/package.py [hunk 3]
```diff
@@ -209,6 +213,26 @@ def hostconfig(self, spec, prefix, py_site_pkgs_dir=None):
...
+        cflags = ' '.join(spec.compiler_flags['cflags'])
+        if cflags:
+            cfg.write(cmake_cache_entry("CMAKE_C_FLAGS", cflags))
+        cxxflags = ' '.join(spec.compiler_flags['cxxflags'])
+        if cxxflags:
+            cfg.write(cmake_cache_entry("CMAKE_CXX_FLAGS", cxxflags))
```
Comment (adrienbernede) [inline scripts/uberenv/packages/serac/package.py:222]: Cool! I like it!

**Comment by white238** [comment]
One more rebuild to fix the mfem package. Adding any flags to the cxx makes it lose opt flags...

**Comment by white238** [comment]
LGTM

**Comment by white238** [comment]
LGTM

**Pull request merged**